polkit: 126 -> 127

skissue opened the pull request:
Fixes: #472582

https://github.com/polkit-org/polkit/blob/main/NEWS.md#polkit-127

Still fairly new to nixpkgs, so let me know if I'm doing anything incorrectly! Not sure what the best way to test this is, still working on rebuilding my system with this change (compilation is brutal).

<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform:
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- Tested, as applicable:
  - [ ] [NixOS tests] in [nixos/tests].
  - [ ] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [ ] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [x] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [x] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

skissue committed:
polkit: 126 -> 127

Fixes: #472582

skissue commented:
Update: I was able to compile `hyprpolkitagent` with an overridden `polkit` input. With it running instead of `polkit-kde-authentication-agent-1`, `pkexec` works :tada:.

Remaining issues:
1. How to handle packages that directly reference the old SUID wrapper?
2. `polkit-agent-helper.socket` is not autostarting (at least on my system). Not sure if this is an issue with not rebuilding everything, or a problem with the unit files.

Sporif commented:
The socket can be autostarted with 
```
systemd.sockets."polkit-agent-helper".wantedBy = [ "sockets.target" ];
```
 `polkit-kde-agent` (and other third-party agents) need to be rebuilt with polkit 127 to get them to work with a suid-less `polkit-agent-helper`.

skissue committed:
nixos/polkit: remove SUID wrapper for polkit-agent-helper-1

Polkit v127 no longer requires a SUID `polkit-agent-helper-1` and instead
recommends using a socket-activated `polkit-agent-helper` service[1].

[1] <https://github.com/polkit-org/polkit/pull/501>

skissue commented:
Changed the target branch from `master` to `staging` according to the contributor guidelines (due to the large number of rebuilds).

LordGrimmauld reviewed (commented):
I had experimented with suid-less systems recently, and also picked the polkit patch enabling the socket activated agent helper. All that was needed was adding the socket to `sockets.target`, and that is being done here. This does look pretty good at a glance. I do plan on adding a suid-less nixos test testing run0, polkit and pam afteri get around to adding `account-utils` as an option. I'll do a more detailed review later today.

LordGrimmauld reviewed (approved):
https://git.grimmauld.de/Grimmauld/grimm-musl/commit/064dd8afcfc41c2083c0b8842b60e99bf6282fb5

<img width="1885" height="477" alt="image" src="https://github.com/user-attachments/assets/72a7454c-6d42-42fd-b08c-a13e2ca6eb52" />

#473403, #453557, #472008, #473068 on musl nixos and current staging, with a couple other fixes

Seems to work, polkit agent requests do work at least for run0. Socket is up, and it even works on musl.
